src/VisualStudio/CSharp/Impl/Options/Formatting/FormattingOptionPageControl.xaml.cs

```diff
@@ -52,5 +90,14 @@ private void SetNestedCheckboxesEnabled()
             FormatOnCloseBraceCheckBox.IsEnabled = FormatWhenTypingCheckBox.IsChecked == true;
             FormatOnSemicolonCheckBox.IsEnabled = FormatWhenTypingCheckBox.IsChecked == true;
         }
+
+        internal void SetCodeCleanupAsConfigured()
+        {
+            // AreCodeCleanupConfiguredCheckBox is hidden all the time, and it tracks if the user ever configured the code cleanup
+            if (!AreCodeCleanupRulesConfiguredCheckBox.IsChecked.HasValue || AreCodeCleanupRulesConfiguredCheckBox.IsChecked.Value == false)
```

Comment:
you can use "CodeCleanupOptions.AreCodeCleanupRulesConfigured" directly?

Reply:
so, if user click "cancel" then we repeatedly show the info bar to users? we might want to split infobar display checking and whether user opt-in to use code clean to 2 different options.

Reply:
I cannot call CodeCleanupOptions.AreCodeCleanupRulesConfigured to get the value directly.

When user click the "config it now" button, the infobar disappears. If user click cancel here, the infobar will not re-appear until user restarts VS.